x/superfluid/keeper/internal/events/emit.go

```diff
@@ -0,0 +1,133 @@
+package events
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v10/x/superfluid/types"
+)
+
+func EmitSetSuperfluidAssetEvent(ctx sdk.Context, denom string, assetType types.SuperfluidAssetType) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newSetSuperfluidAssetEvent(denom, assetType),
+	})
+}
+
+func newSetSuperfluidAssetEvent(denom string, assetType types.SuperfluidAssetType) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtSetSuperfluidAsset,
+		sdk.NewAttribute(types.AttributeDenom, denom),
+		sdk.NewAttribute(types.AttributeSuperfluidAssetType, assetType.String()),
+	)
```

Comment:
Since we are going over all the module events, what are the thoughts on using `EmitTypedEvent` instead of `EmitEvents`

It'll definitely reduce the code size and gives a better clarity. For instance the above code could be rewritten as with an event.proto file; 
```
ctx.EventManager().EmitTypedEvent(&superfluid.AssetEvent{
		denom:  denom,
	         assetType: assetType.String(), 
	})
```

ps: `EmitEvents` is labelled deprecated too

cc @alexanderbez @ValarDragon @p0mvn

Reply:
Yes, I think that's a great idea. We would have to make the event constructors exported.

For example, `newSetSuperfluidAssetEvent` -> `NewSetSuperfluidAssetEvent`

Reply:
Are we reusing events? I would only use constructors and separate out the event logic if we use the same ones in multiple places, otherwise it would be overkill.

Reply:
We are in 0 rush to switch to typed events. 

I also agree that constructors seem overkill